### DISCUSSION TITLE
Crash when working with navigation api and NavigationSplitView

### DISCUSSION THREAD

**Discussion by marcuswu0814** [pointfreeco/swift-composable-architecture#2915] [Q&A]
Hi @mbrandonw, this is my [reproduce project](https://github.com/marcuswu0814/TCANavigationSplitView) that crash when building my features with `NavigationSplitView` and TCA navigation api (maybe cause by NavigationPath).

The project:
* Root with a `NavigationSplitView(sideBar:detail:)` two columns mode
* Three tab, A and B build with a `NavigationStack(path:)` api, and using the pattern like `Reducer(state: .equatable) enum Path {}`
* C Tab build with a pure `Text("...")`

Problems:
* If switch A -> C -> B, just ok
* Switch from A -> B directly, crash. `SwiftUI/NavigationColumnState.swift:560: Fatal error: 'try!' expression unexpectedly raised an error: SwiftUI.AnyNavigationPath.Error.comparisonTypeMismatch`

I'm not sure it was a best practice to use `NavigationSplitView` with TCA like my project, if a better solution exist, please give some advice for me. 🙏

**Comment by marcuswu0814** [comment]
I try to add a [workaround](https://github.com/marcuswu0814/TCANavigationSplitView/tree/workaround) for the crash.

I replace the `NavigationLink(title:value:)` with a wrapper of a pure `Button` like:

```swift
@ViewBuilder
func navigationLink(
    _ title: String,
    value: Feature.State.Tab
) -> some View {
    Button(title) {
        store.send(.switchTab(value))
    }
    .tag(value)
}
```

and mutating the selection state via action, the crash just gone, but still want to know how to fix the issue and way to work `NavigationSplitView` more elegance with TCA.

**Reply by mbrandonw** [reply]
Hi @marcuswu0814, I don't think `NavigationSplitView` is meant to be used in the way you have structured things above. Each of your "tab" views have their own `NavigationStack`, and that isn't correct. There should only be one single `NavigationStack` in the `detail` trailing closure:

```swift
NavigationSplitView {
  …
} detail: {
  NavigationStack(path: …) {
    …
  }
}
```

I recommend looking at Apple's [Cookbook](https://developer.apple.com/documentation/swiftui/focus-cookbook-sample) demo app for more information on how to properly use `NavigationSplitView`.
